proto/backend.proto

```diff
@@ -306,3 +306,164 @@ message StreamPacket {
   // JSON-encoded data to publish into a channel.
   bytes data = 1;
 }
+
+//-----------------------------------------------------------------
+// InstanceSettings -- EXPERIMENTAL and is subject to change until 12.??
+//-----------------------------------------------------------------
+
+service InstanceSettings {
```

Comment:
Naming could be improved I think. Main question whether to continue to keep protocol/SDK generic without k8s specifics or not. You mentioned upstream k8s proto, I'm not sure that's a path we want to take, but open for discussion.

Related to other comment about explicit and as a comparison. Not sure this makes more sense, but an alternative to the operation enum and plugin context only available on update/delete. In addition, if existing plugin context provided for update/delete it follows the existing services/rpc's and could make use of the instance caching and to me implies that the updated instance settings is provided as a separate field for updates. This separate field exists for create as well, but not for deletes.

```proto
service InstanceMutation {
  rpc CreateInstance(CreateInstanceRequest) returns (...)
  rpc UpdateInstance(UpdateInstanceRequest) returns (...)
  rpc DeleteInstance(DeleteInstanceRequest) returns (...)
}
```

Reply:
🤔 interesting ya this could be nicer -- honestly I was scared off by how much boilerplate there is!

This gives us a nice way to have the [previous values included for update ](https://github.com/grafana/grafana-plugin-sdk-go/pull/983#discussion_r1606352201)(it is just pluginContext!).

Reply:
We all love boilerplate, right 😄 

> This gives us a nice way to have the [previous values included for update ](https://github.com/grafana/grafana-plugin-sdk-go/pull/983#discussion_r1606352201)(it is just pluginContext!).

Yes, and I guess the check health stuff is maybe only relevant for create? Or maybe not. But I guess the point is that you might not be able to just reuse the existing check health handler for create because it's normally dependent on executing in an existing (cached) instance where the struct has been initialized custom for each plugin with the `InstanceFactoryFunc`.

Reply:
If you want to experiment with the API and as mentioned, given proto allows all request/response payloads you can keep the existing rpc definition and split things up into multiple handlers in the SDK if that simplify/removes some boilerplate and then later when things have stabilized we can change the protobuf if needed/makes sense. At least maybe allow to iterate faster?

Reply:
I took a stab at splitting these into individual functions... it seems like a good idea!

BUT in the end it looks like a lot of boilerplate that will all likely point to the same function anyway.  There is likely a reason the k8s hooks all point to the same object and use operation to specify the what verb was requested.

See for example: https://github.com/grafana/grafana/compare/plugin-validate-settings-no-operation?expand=1#diff-5c561a6f4b7dfc53ddc8dda6337a86a03e24e9dc51720314b37d7fb2259efa80R28

Reply:
Okay I see